frontend/packages/console-shared/src/hooks/useBuildConfigsWatcher.ts

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { CronJobKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { getBuildConfigsForCronJob, getBuildConfigsForResource } from '../utils';
+import { BuildConfigOverviewItem } from '../types';
+
+export const useBuildConfigsWatcher = (
+  resource: K8sResourceKind,
+): { loaded: boolean; loadError: string; buildConfigs: BuildConfigOverviewItem[] } => {
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>('');
```

Comment:
Please default with null. The useEffect also sets setLoadError to null if there is new resources available without any error. In this case we can save us from one re-rendering